### DISCUSSION TITLE
Tentative Roadmap for next 2 weeks as of 3rd May

### DISCUSSION THREAD

**Discussion by rgxai** [covidarmy/backend#35] [General]
# Introduction

Hello everyone, first of all, I would really like to appreciate and thank all of you for taking out your valuable time and helping with your valuable contributions. I am jotting down my thoughts on what our focus areas for the coming 2 weeks should be and how I see us playing an integral role in minimising loss of valuable lives by providing timely assistance to those looking for COVID related help. Please share your thoughts and feedback on the below pointers.

# Problems

I have observed the following problems in the current systems some of which we can take up and solve

1. Lack of (near) real-time and verified consolidated backend data of suppliers of COVID essentials across 1400+ cities
2. Lack of an easy to use pan India interface to find COVID related help quickly
3. Suppliers are spending most of their time answering calls instead of focusing on improving their capacity. This leads to the supply nodes being shut down or switched off because they are unable to handle so many requests.
4. Lack of proper tele-health guidance and assistance required for treating mild and moderate cases at home. There is a lot of misinformation on the use of remdesivir, etc.
5. Lack of an easy to use mechanism to volunteer for verification of contact numbers
6. Lack of an easy to use mechanism for volunteers to find people to help
6. Blacklisting fraud and scamsters
7. Making vaccine supply chain efficient. Help people in finding vaccination slots nearby and reducing vaccine wastage. Currently over 10% of vaccine doses are wasted as they need to be used within 4 hours of opening! Read [here](https://www.hindustantimes.com/cities/delhi-news/vaccine-wastage-at-4-1-for-covidshield-17-5-for-covaxin-in-delhi-101612893597345.html).
8. Data analysis of demand to evaluate which city needs the most oxygen cylinders, concentrators, hospital beds, etc.

# Solutions
The proposed solution is a combination of multiple things which will cohesively help us build a strong 

1. Real time parsed data stream from Twitter (and in the future other sources such as Telegram, OCR, etc) which gets us data of resources as per the following [schema](https://www.notion.so/catoverse/New-Database-Schema-b3c392d452ba47b0a5ec0e05804edd90)
2. Multi-Lingual (initially English and Hindi) Whatsapp Chatbot to access the above data easily. The chatbot will also have flows for Volunteers who can volunteer to add new leads or verify existing leads in our database and a flow for Suppliers who can register their services and get matched to people looking for help with that specific service. The work in progress flow of the bot is [here](https://www.figma.com/file/CydTsdqTthlvRrREsggeCC/IND-COVID-BOT-Flow-Final?node-id=0%3A1)
3. Website to filter real time leads by city and requirement type and get in touch with the supplier (covid.army). The website will also have details about homecare COVID practices and useful COVID related knowledge base such as proning.
4. Load balancing of demand to suppliers so that a single supplier is not over flooded. We need a strong backend that can take in inputs from multiple sources (such as chatbot interactions, tweet replies, website inputs, etc) to change the verification status of the leads and accordingly divide it between users and volunteers
5. District wise demand dashboard - data estimated based on Twitter demand leads (or Whatsapp chatbot requests) where civic bodies can analyse the need for oxygen concentrators, ICU beds, etc
6. Vaccine alert - Register on website for a notification in your District and get alerted when vaccine slots open up (prototype already built by Utkarsh [here](https://cowinapp.workofutkarsh.com/))
7. A dashboard where Volunteers can easily see requests for help and get in touch with them to help them out

# Other pointers

1. Ensure proper documentation of our code and issues on GitHub so that we can roll out the repos to more volunteers to collaborate
2. Distribution - This is something that most platforms struggle with but fortunately, through our extended list of partners and people network, we should be able to achieve significant traction if we design the above solutions properly

**Comment by vsengar26** [comment]
We should also implement the WhatsApp feature this week to enable user reach out more resources in the same amount of time. It's super easy to implement, just a regex and a hyperlink. Having predefined texts will also save a lot of time typing same text to various persons. Also, it'll allow to message anyone without saving his number.

**Reply by rgxai** [reply]
Agreed. We can show call and whatsapp icons near the lead. Thoughts?

**Reply by vsengar26** [reply]
Will work but if possible we should have a design that on clicking the WhatsApp/Call icon, a small popup appear (as in that appears on our phone when we select any texts that has options to copy/share etc) which contains all the numbers present in tweet. 

Since many tweets contain several numbers, we should facilitate options to send messages to each of them.

**Reply by rgxai** [reply]
We will be breaking down tweets containing multiple numbers into separate DB entries for simplicity

**Reply by brainless** [reply]
I know the discussion about the WhatsApp bot in the main post. But I do not understand the WhatsApp feature you are suggesting @vsengar26. Perhaps I was not in the discussion or am confused. Can you please elaborate on this?

**Reply by rgxai** [reply]
@brainless he is talking about showing a whatsapp button on the covidarmy website where the user can click and is directly taken to a whatsapp chat with that number with a "Hi" prompt. This doesn't have anything to do with the chatbot.

**Reply by vsengar26** [reply]
It isn't about WhatsApp bot. Its a different thing. I suggested that we should parse the tweets to get mobile numbers present in them. Then we can have an WhatsApp icon which on clicking will open WhatsApp web or app depending on user is using through mobile or web.

Upon opening it will has a pre-written message and user has to just send.

For example suppose tweet contain a number say X. Then we'll parse this X using regex and when someone clicks on WhatsApp icon, his WhatsApp will be opened with a message like Hey I need this this resource, written to X and he has to just send it.

**Reply by samanyougarg** [reply]
That sounds like a great feature usability wise and shouldn't be hard to implement.

**Reply by rgxai** [reply]
Yup please create an issue for this and we can take this up once we have parsed data coming in.
